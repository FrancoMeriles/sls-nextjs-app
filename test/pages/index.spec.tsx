import { render, screen } from '@testing-library/react'
import App from '@base/pages'
import renderer from 'react-test-renderer'

jest.mock('@base/utils', () => ({
  getRandomColor: () => 'hsl(214,100%,50%)',
}))

const mockCategories = [
  { id: 1, name: '__FIRST_NAME__' },
  { id: 2, name: '__SECOND_NAME__' },
]

describe('App', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<App categories={mockCategories} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should renders and display two categories with the correct name and url', async () => {
    render(<App categories={mockCategories} />)
    mockCategories.forEach((mockCategorie) => {
      expect(screen.getByText(mockCategorie.name)).toBeInTheDocument()
      expect(screen.getByText(mockCategorie.name).closest('a')).toHaveAttribute(
        'href',
        `/categorie/${mockCategorie.id}/products?page=1`
      )
    })
  })
})
