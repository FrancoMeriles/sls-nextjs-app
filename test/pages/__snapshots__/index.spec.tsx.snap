// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render and match snapshot with required props 1`] = `
<div
  className="chakra-container css-f91va8"
>
  <div
    className="css-s13uex"
  >
    <div
      className="chakra-linkbox css-15wvuzs"
    >
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/categorie/1/products?page=1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <p
          className="chakra-text css-pelz90"
        >
          __FIRST_NAME__
        </p>
        <span
          className="css-1ksdfwg"
        />
      </a>
    </div>
    <div
      className="chakra-linkbox css-15wvuzs"
    >
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/categorie/2/products?page=1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <p
          className="chakra-text css-pelz90"
        >
          __SECOND_NAME__
        </p>
        <span
          className="css-1ksdfwg"
        />
      </a>
    </div>
  </div>
</div>
`;
